File: pyez/basic-example/basic.py
'''
Do selective imports from Python Packages

from x import y #imports only the y package from the greater package set of x
'''
from pprint import pprint
from jnpr.junos import Device

'''
Create a new device object. This represents the SRX we are going to connect to.

The hostname/IP, username, and password are required.
'''
dev = Device(host='172.23.230.204', user='root', password='juniper123' )

'''
Creates a Netconf session to the SRX
'''
dev.open()

'''
Prints out the device facts in a pretty formated way
'''
pprint( dev.facts )

'''
Closes the netconf connection to the SRX
'''
dev.close()

File: rubyez/basic-example/basic.rb
#Require the inital packages that are required for this tool    
require 'pp'                #Allows for pretty printing of output
require 'net/netconf/jnpr'  #Allows for netconf connection
require 'junos-ez/stdlib'   #Library to simplify junos management
require 'junos-ez/srx'      #Library to simplify srx management

# login information for NETCONF session 
#
login = { :target => '172.21.203.91', :username => 'root',  :password => 'juniper123',  }

#create a connection object
dev = Netconf::SSH.new( login )

print "Connecting to device #{login[:target]} ... "
#open connection to SRX
dev.open()
puts "OK!"

#Bind provider to our connection
#This basic provider gives us facts about the device
Junos::Ez::Provider( dev )

#Prints out the device facts in a pretty formated way
pp dev.facts.catalog

#Close the netconf connection to the SRX
dev.close
